db/eventlog.c

```diff
@@ -454,22 +454,8 @@ static void eventlog_add_newsql(cson_object *obj, const struct reqlogger *logger
     cson_value_free(newval);
 }
 
-static void eventlog_add_int(cson_object *obj, const struct reqlogger *logger)
+static void populate_obj(cson_object *obj, const struct reqlogger *logger)
 {
-    Pthread_mutex_lock(&eventlog_lk);
-    if (eventlog == NULL || !eventlog_enabled) {
-        Pthread_mutex_unlock(&eventlog_lk);
-        return;
-    }
-
-    bool isSql = logger->event_type && (strcmp(logger->event_type, "sql") == 0);
-    bool isSqlErr = logger->error && logger->stmt;
-
-    if ((isSql || isSqlErr) && !hash_find(seen_sql, logger->fingerprint)) {
```

Comment:
Moved the adding eventlog_add_newsql() to the caller, while it already has the lock so no need to get/release lock here.

Reply:
gotcha